src/Hub/Middleware/CorsMiddleware.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Freddie\Hub\Middleware;
+
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use React\Http\Message\Response;
+
+class CorsMiddleware
+{
+    public function __construct(
+        private ?string $corsOrigin
+    )
+    {
+        if (!$this->corsOrigin) {
+            $this->corsOrigin = 'null';
+        }
+    }
```

Comment:
I would do this instead:

```php
public function __construct(
    private readonly ?string $corsOrigin = '*',
) {}
```

Then, in `__invoke()`,  if `$this->corsOrigin` is `null`, coalesce to `$this->getOrigin($request)` and implement `getOrigin` so that it parses the `Origin` or `Referer` header.

Why?
By default, '*' will be responded to the browser, meaning "trust any origin", except it won't be compatible with `Access-Control-Allow-Credentials: true` (which should be added, to allow the `mercureAuthorization` cookie to work properly when used). If the developer doesn't specify any origin, then it would mean the middleware would have to parse the request's origin and respond that this origin is allowed.